04-data-structures-part1.Rmd

```diff
@@ -339,125 +259,110 @@ In this case, we've explicitly told R that "control" should represented by 1, an
->
-
-Lists are extremely useful inside functions. You can "staple" together lots of
-different kinds of results into a single object that a function can return. In
-fact many R functions which return complex output store their results in a list.
+and we can ask for and put values in the elements of our matrix with a couple of different notations:
 
-## Challenge solutions
+```{r}
+x[1,1] = 1
+x
+x[1][1]
+x[1][1] = 2
+x[1,1]
+x
+```
 
-> ## Solution to challenge 1: Data types {.challenge}
 >
-> Use your knowledge of how to assign a value to
-> a variable, to create examples of data with the
-> following characteristics:
->
-> 1) Variable name: 'answer', Type: logical
-> 2) Variable name: 'height', Type: numeric
-> 3) Variable name: 'dog_name', Type: character
+> ## Challenge 3 {.challenge}
 >
-> For each variable you've created, test that it
-> has the data type you intended. Do you find
-> anything unexpected?
+> What do you think will be the result of
+> `length(x)`?
+> Try it.
+> Were you right? Why / why not?
 >
-> ```{r}
-> answer <- TRUE
-> height <- 150
-> dog_name <- "Snoopy"
-> is.logical(answer)
-> ```
+
 >
-> ```{r}
-> is.numeric(height)
-> ```
+> ## Challenge 4 {.challenge}
 >
-> ```{r}
-> is.character(dog_name)
-> ```
+> Make another matrix, this time containing the numbers 1:50,
+> with 5 columns and 10 rows.
+> Did the `matrix` function fill your matrix by column, or by
+> row, as its default behaviour?
+> See if you can figure out how to change this.
+> (hint: read the documentation for `matrix`!)
 >
 
-> ## Solution to challenge 2 {.challenge}
+> ## Challenge 5 {.challenge}
+> Create a list of length two containing a character vector for each of the sections in this part of the workshop:
 >
-> Vectors can only contain one atomic type. If you try to combine different
-> types, R will create a vector that is the least common denominator: the
-> type that is easiest to coerce to.
+> - Data types
+> - Data structures
 >
-> ```{r}
-> xx <- c(1.7, "a")
-> xx
-> typeof(xx)
-> ```
+> Populate each character vector with the names of the data types and data structures we've seen so far.
 >
-> ```{r}
-> xx <- c(TRUE, 2)
-> xx
-> typeof(xx)
-> ```
+
+
+
+
+
+
+
+
+## Challenge solutions
+
+> ## Discussion 1 {.challenge}
+> By keeping everything in a column the same, we allow ourselves to make simple assumptions about our data;
+> if you can interpret one entry in the column as a number, then you can interpret *all* of them as numbers,
+> so we don't have to check every time. This consistency, like consistently using the same separator in our 
+> data files, is what people mean when they talk about *clean data*; in the long run, strict consistency goes 
+> a long way to making our lives easier in R. 
 >
+
+> ## Solution to Challenge 1 {.challenge}
 > ```{r}
-> xx <- c("a", TRUE)
-> xx
-> typeof(xx)
+> x <- [11:20]
+> firstFive <- head(x, 5)
+> thirdToFifth <- tail(firstFive, 3)
+> names(thirdToFifth) <- c('S', 'W', 'C')
 > ```
```

Comment:
This is a strange way to answer this challenge, since square brackets and sequences have been introduced `x[3:5]` works just as well and could be added as an alternate answer.